blog/2023-07-12-angular-tutorial.md

```diff
@@ -0,0 +1,614 @@
+
+```bash
+ng new tolgee-ngx
+```
+
+You will be prompted to provide some specifications for your project.
+
+```
+? Would you like to add Angular routing? Yes
+? Which stylesheet format would you like to use? (Use arrow keys)
+> CSS
+  SCSS   [ <https://sass-lang.com/documentation/syntax#scss>                ]
+  Sass   [ <https://sass-lang.com/documentation/syntax#the-indented-syntax> ]
+  Less   [ <http://lesscss.org>                                             ]
+```
+
+```bash
+cd tolgee-ngx
+ng serve --open
+```
+
+This command launches your app, watches your files and rebuilds it whenever you make changes.
+The `--open` flag automatically opens your browsers to the `[localhost:4200](http://localhost:4200/)`.
+
+![Angular Application](/img/blog/angular-tutorial/angular_app.png)
+
+This is the basic application we will be translating with Tolgee. We will add a select component that will enable us to switch from one language to another.
+
+## Step 4: Install Tolgee
+
+To translate your angular application we need to install the Tolgee package. Open another terminal window and run the command below in the project’s root directory.
+
+```bash
+npm i @tolgee/ngx
+```
+
+The command will install all the necessary dependencies.
+
+## Step 5: Add Environment Variables
+
+In the `src` directory create `environments/environment.local.ts` file and paste the following code:
+
+```typescript
+export const environment = {
+  production: false,
+  tolgeeApiUrl: 'Your server url here',
+  tolgeeApiKey: 'Your api key here',
+};
+```
+
+Replace `tolgeeApiUrl` with your server URL and `tolgeeApiKey` with the API key you generated earlier. Since this app is not for production, we set this variable to false.
+
+## Step 6: Setup Tolgee NgModule
+
+Open your `src/app/app.module.ts` and add the following code:
+
+```typescript
+...
+
+import {
+  DevTools,
+  NgxTolgeeModule,
+  Tolgee,
+  TOLGEE_INSTANCE,
+  FormatSimple,
+} from '@tolgee/ngx';
+import { environment } from 'src/environments/environment.local';
+
+...
+
+@NgModule({
+  ...
+  imports: [..., NgxTolgeeModule],
+
+  providers: [
+    {
+      provide: TOLGEE_INSTANCE,
+      useFactory: () => {
+        return Tolgee().use(DevTools()).use(FormatSimple()).init({
+          language: 'en',
+          apiUrl: environment.tolgeeApiUrl,
+          apiKey: environment.tolgeeApiKey,
+        });
+      },
+    },
+  ],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+Add the `NgxTolgeeModule` in the import section and provide a Tolgee instance in the providers. Our default application language is en you can change it to fr or es depending on your preferences. Lastly, we import the environment variable we added earlier which will help us to communicate with the server.
+
+- The DevTools() is used for in-context translating in dev mode
+- The FormatSimple() enables you to pass variables into translations
+- The BackendFetch() fetches data from the custom backend
```

Comment:
You actually don't use the BackendFetch plugin and you don't need it in most cases. Only thing the plugin does is that if you published your i18n data for example on `https://example.com/i18/en.json`, etc., it can fetch the data from there. But it's not something you should cover in this article.